ttnn/cpp/ttnn/distributed/distributed_tensor_config.cpp

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <unordered_map>
+#include <string>
+
+#include "common/assert.hpp"
+#include "ttnn/distributed/distributed_tensor_config.hpp"
+
+namespace tt::tt_metal {
+namespace {
+
+DistributedTensorConfig create_shard_distributed_tensor_config(
+    const std::unordered_map<std::string, std::string>& metadata) {
+    return ShardTensor(std::stoi(metadata.at("shard_dim")));
```

Comment:
This is a user facing API. I'm not personally a fan of accessing user populated data structures without validation. If a user incorrectly populates the map, this would result in an unordered_map out of bounds error which is not very descriptive.
Additionally, I'm wondering why the values are represented as strings. Would make more sense if its an integer. Is this for future proofing?

Reply:
Agree, definitely not ideal. In this PR I am just scoping this out in its own header to re-structure the file organization. Following up with the API change is a follow up.